### PR TITLE
chore: separate traffic_light_utils from perception_utils

### PR DESCRIPTION
**Pull request by miursh**
## Description

I separated "traffic_light_utils" and "object_recognition_utils" from "perception_utils". 
The main reason for this is to reduce dependencies for the build of "tensorrt_yolox" in the "Edge.auto" project, which relies on new "object_recognition_utils".

## Tests performed
I have confirmed that whole traffic_light_recognition module and object_recognition module work correctly after this PR by logging_simulator.launch.
image
image

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@miursh May I ask for some background on this PR?
I feel uncomfortable that only traffic light is separated from perception utils.

**Review by yukkysaito — CHANGES_REQUESTED** [review]
@miursh May I ask for some background on this PR?
I feel uncomfortable that only traffic light is separated from perception utils.

**Comment by yukkysaito** [comment]
@miursh cc @mitsudome-r 
Thanks for the updated PR description.
If you want to separate them, I'd like to go from perception_utils to object_recognition_utils.

**Comment by miursh** [comment]
@yukkysaito 
Thank you for your comment.
Do you mean separating into 'traffic_light_utils' and 'object_recognition_utils' instead of 'perception_utils'?

**Comment by yukkysaito** [comment]
>Thank you for your comment.
Do you mean separating into 'traffic_light_utils' and 'object_recognition_utils' instead of 'perception_utils'?

@miursh Yes.
If it is a common use of object_recognition and traffic_light, then perception_utils is fine.

**Comment by miursh** [comment]
@yukkysaito I have renamed the most of perception_utils to object_recognition_utils.
https://github.com/autowarefoundation/autoware.universe/pull/4207/commits/d7a8c57a8a1e79f2b5b129bf0772082d3f948112

**Review by yukkysaito — APPROVED** [review]
LGTM

File: common/traffic_light_utils/include/traffic_light_utils/traffic_light_utils.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 #include <tf2/LinearMath/Matrix3x3.h>
 #include <tf2/LinearMath/Transform.h>
 
-namespace perception_utils
+namespace traffic_light_utils
 {
 
 namespace traffic_light
```
Comment (wep21) [inline common/traffic_light_utils/include/traffic_light_utils/traffic_light_utils.hpp:30]: @miursh This namespace seems redundant if the first namespace is changed from `perception_utils` to `traffic_light_utils`

File: common/perception_utils/include/perception_utils/prime_synchronizer.hpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ namespace perception_utils
 {
 
 /**
- * @brief  This class implemented a multi-topic approximate time synchronizer.
+ * @brief  This class implemented a multi-topic approxiamate time synchronizer.
```
Comment (wep21) [inline common/perception_utils/include/perception_utils/prime_synchronizer.hpp:33]:
```suggestion
 * @brief  This class implemented a multi-topic approximate time synchronizer.
```

**Review by yukke42 — APPROVED** [review]
LGTM

**Pull request merged**